boards/FSM/Src/Io/Io_FlowMeter.c

```diff
@@ -1,20 +1,91 @@
 #include <assert.h>
 #include "main.h"
-#include "Io_SharedPwmInput.h"
+#include "Io_SharedFrequencyOnlyPwmInput.h"
 #include "Io_FlowMeter.h"
 
-static struct PwmInput *  primary_flow_meter_pwm_input;
-static TIM_HandleTypeDef *primary_flow_meter_htim;
+#define PRIMARY_FLOW_METER_CHANNEL HAL_TIM_ACTIVE_CHANNEL_1
+#define SECONDARY_FLOW_METER_CHANNEL HAL_TIM_ACTIVE_CHANNEL_2
+
+extern TIM_HandleTypeDef htim4;
+
+static struct FrequencyOnlyPwmInput *primary_flow_meter_pwm_input,
+    *secondary_flow_meter_pwm_input;
+static TIM_HandleTypeDef *flow_meters_timer_handle;
+
+static size_t primary_channel_period_elapsed_tick   = 0U;
+static size_t secondary_channel_period_elapsed_tick = 0U;
+
+void Io_FlowMeter_Init(void)
+{
+    // Configuring timer handle for both flow meters
+    flow_meters_timer_handle = &htim4;
+
+    primary_flow_meter_pwm_input = Io_SharedFrequencyOnlyPwmInput_Create(
+        &htim4, TIMx_FREQUENCY / TIMx_PRESCALER, TIM_CHANNEL_1);
+    secondary_flow_meter_pwm_input = Io_SharedFrequencyOnlyPwmInput_Create(
+        &htim4, TIMx_FREQUENCY / TIMx_PRESCALER, TIM_CHANNEL_2);
+}
 
 float Io_FlowMeter_GetPrimaryFlowRate(void)
 {
-    return Io_SharedPwmInput_GetFrequency(primary_flow_meter_pwm_input) / 7.5f;
+    return Io_SharedFrequencyOnlyPwmInput_GetFrequency(
+               primary_flow_meter_pwm_input) /
+           7.5f;
+}
+
+float Io_FlowMeter_GetSecondaryFlowRate(void)
+{
+    return Io_SharedFrequencyOnlyPwmInput_GetFrequency(
+               secondary_flow_meter_pwm_input) /
+           7.5f;
+}
+
+void Io_FlowMeter_Primary_FlowMeter_Tick(void)
```

Comment:
nit; Can we keep the function names more consistent as in `{App|Io}_{ModuleName}_{Description}`. So this one would be `Io_FlowMetesr_TickPrimary` (Removed the flow meter part cuz it seemed redundant)

Reply:
Fixed